common/prometheus-server/templates/alerts/_prometheus.alerts.tpl

```diff
@@ -376,3 +376,20 @@ groups:
         Prometheus `{{`{{ $labels.prometheus }}`}}` remote write is behind for `{{`{{ $labels.url }}`}}`.
       summary: Prometheus remote write is behind.
   {{- end }}
+  {{- if eq $root.Values.vpaUpdateMode "Auto" }}
+  - alert: PrometheusVPARecommendsVPAMaxAllowedExceeded
+    expr: vpa_butler_vpa_container_recommendation_excess{verticalpodautoscaler=~"prometheus.+"} / 1024 / 1024 / 1024 > 0
```

Comment:
As this is a common chart, we need to tie the alert to the release in question to avoid receiving multiple unrelated alerts.

Reply:
Can you elaborate a little more? We are only configuring the alert if it is set to `Auto`, which is not the case yet. When we're enabling respective Prometheis the set to `Auto` will appear at the same time as the version bump.

Reply:
I see. 

Three things, imagine we roll this out with e.g. prometheus-openstack
1. the metric `vpa_butler_vpa_container_recommendation_excess` does not exist. Leads to an absent-metric alert
2. if this is depoyed with the corresponding Thanos Rule, multiple prometheus are in scope - not just prometheus-openstack. In other words: We get an alert deployed with prometheus-openstack also for other prometheus.
3. the metric also has `resource="cpu", unit="core"` labels - so this does not only apply to memory

Reply:
1. and 2.: 
You're right about the thanos rule, since prometheus-kubernetes was the testing horse I haven't had this in mind.
We got to move this accordingly.

3. Yes, the calculation makes no sense in this case. It was just for better readability. Initially I thought of handling this in one alert, but the GiB conversion makes no sense then. We can split it in two, and have better readable Memory and CPU values.